### PR TITLE
Solve sim speed

### LINKED ISSUE #76: Run simulations at different speeds
## Description
Right now a simulation run happens "in real time"/"at the same speed as the original trace".  It would be useful to be able to run a simulation at a constant-factor speedup (probably based on a CRD/config setting somewhere).  There can be some risks/inaccuracies doing this, especially if anything autoscaling is involved, but if you have a four hour trace from an incident you probably don't want to sit around for four hours every time you replay it.

[Code hint](https://github.com/acrlabs/simkube/blob/master/sk-driver/src/runner.rs#L162)

### PR DESCRIPTION
**Pull request by GeraldoSJr**
- [x] I certify that this PR does not contain any code that has been generated with GitHub Copilot or any other AI-based code generation tool, in accordance with this project's policies.

## Description
This PR fix the issue Run simulation at different speeds by adding a argument, sim_step_duration, to the cli and passing this argument to the run_trace function and allowing the simulation to run on this speed instead of the fixed timestamp speed specified on the trace.

## Testing done
- How did you test your changes? Manually deployed the code and ran simulation with different parameters and speeds to measure the impact of the fix.
- Does this code require any new tests to be written? No.

## Additional info
Resolves #76

### CODE REVIEW

**Comment by drmorr0** [comment]
## Updating Kubernetes DAG...
Please wait until the job has finished.

**Review by drmorr0 — CHANGES_REQUESTED** [review]
Hi, @GeraldoSJr, thank you for the PR!  Can you please update code changes so that instead of using a "stepwise" function, the speed is controlled by a constant factor?  I.e., 

```
let sleep_duration = max(0, (next_ts - sim_ts) / speed_factor);
```

This way we're guaranteed that, with a speed factor of 2, that the simulation will run exactly twice as fast.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .build
 .vendor
 generated-api
+Makefile
```
Comment (drmorr0) [inline .gitignore:4]: Makefile/build shouldn't be in .gitignore

**Comment by GeraldoSJr** [comment]
Added the changes requested, that was actually my innitial approach, changed midway to resolve the issue more directly, anyways, hope this will solve it now. Just wanted to say that I'm excited to start helping in the repo, really nice project! 👍

File: sk-driver/src/runner.rs [hunk 2]
```diff
@@ -162,7 +162,16 @@ pub async fn run_trace(ctx: DriverContext, client: kube::Client) -> EmptyResult
...
+            let simulation_normal_step_duration = next_ts - sim_ts;
+            let simulation_normal_step_duration_u64 = simulation_normal_step_duration.try_into().unwrap_or(u64::MAX);
+
+            let sleep_duration = if speed <= 0.0 {
+                simulation_normal_step_duration_u64
+            } else {
+                (simulation_normal_step_duration_u64 as f64 / speed).round() as u64
+            };
+
+            let sleep_duration = max(0, sleep_duration);
```
Comment (drmorr0) [inline sk-driver/src/runner.rs:174]: I think we can simplify this whole thing significantly:

```
let simulation_normal_step_duration = max(0, next_ts - sim_ts) as f64;
let sleep_duration = (simulation_normal_step_duration / speed).round() as u64;
```

That way we ensure that even if there's a negative step duration, that event will happen "right now"; and speed should never be negative at this point (ideally we'd have some validation here, but that can be done in a different PR).

**Review by drmorr0 — CHANGES_REQUESTED** [review]
Hi -- a few small changes, but this is basically there.  Thanks!

**Review by drmorr0 — APPROVED** [review]

**Pull request merged**